### PR TITLE
Jonmv/expanded zk api

### PR DESCRIPTION
**Pull request by jonmv**
@hakonhall please review.

### CODE REVIEW

File: zkfacade/src/test/java/com/yahoo/vespa/curator/CuratorWrapperTest.java [hunk 1]
```diff
@@ -93,11 +97,18 @@ public void testSingleSingleton() {
             // ... and deactivated as a result of unregistering again.
 
             // Singleton can be set up again, but this time, time runs away.
+            Phaser mark2 = new Phaser(2); // Janitor and helper.
+            new Thread(() -> {
+                mark2.arriveAndAwaitAdvance(); // Wait for janitor to call activate.
+                stunning.arriveAndAwaitAdvance(); // Let janitor measure time spent on activation, while test thread waits for it.
+                stunning.arriveAndAwaitAdvance(); // Let janitor measure time spent on activation, while test thread waits for it.
+            }).start();
```
Comment (jonmv) [inline zkfacade/src/test/java/com/yahoo/vespa/curator/CuratorWrapperTest.java:105]: 🙈 🙊 🙉

**Comment by jonmv** [comment]
"something goes wrong elsewhere" is more about future-proofing the code. I don't think it has any effect _right now_ :p

**Review by hakonhall — APPROVED** [review]

**Comment by jonmv** [comment]
Still some trouble here ... :>

File: document/src/main/java/com/yahoo/document/json/package-info.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
```
Comment (hakonhall) [inline document/src/main/java/com/yahoo/document/json/package-info.java:1]: Accidental removal?

Comment (jonmv) [reply document/src/main/java/com/yahoo/document/json/package-info.java:1]: Definitely.

Comment (jonmv) [reply document/src/main/java/com/yahoo/document/json/package-info.java:1]:
```suggestion
// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
```

**Review by hakonhall — APPROVED** [review]

**Pull request merged**